From: Tzafrir Cohen <tzafrir@debian.org>
Subject: Fix building chan_phone on big endian systems
Bug: https://issues.asterisk.org/jira/browse/ASTERISK-24458

--- a/channels/chan_phone.c
+++ b/channels/chan_phone.c
@@ -827,7 +827,7 @@ static int phone_write(struct ast_channe
 		} else {
 			int swap = 0;
 #if __BYTE_ORDER == __BIG_ENDIAN
-			if (frame->subclass.format.id == AST_FORMAT_SLINEAR)
+			if (ast_format_cmp(frame->subclass.format, ast_format_slin) == AST_FORMAT_CMP_EQUAL)
 				swap = 1; /* Swap big-endian samples to little-endian as we copy */
 #endif
 			res = phone_write_buf(p, pos, expected, maxfr, swap);
