Description: set PATH_MAX if not set (HURD)
Author: Gabriele Giacone <1o5g4r8o@gmail.com>
Bug: https://issues.asterisk.org/jira/browse/ASTERISK-25070
Last-Update: 2016-04-02

--- a/include/asterisk.h
+++ b/include/asterisk.h
@@ -320,4 +320,8 @@ struct ast_sched_context;
  */
 int ast_pbx_uuid_get(char *pbx_uuid, int length);
 
+#ifndef PATH_MAX
+# define PATH_MAX 4096
+#endif
+
 #endif /* _ASTERISK_H */
